libs/dal/src/repositories/environment/environment.repository.ts

```diff
@@ -87,4 +88,23 @@ export class EnvironmentRepository extends BaseRepository<EnvironmentDBModel, En
 
     return await this.getApiKeys(environmentId);
   }
+
+  async updateApiRateLimits(environmentId: string, apiRateLimits: Partial<IApiRateLimitMaximum>) {
+    await this.update(
+      {
+        _id: environmentId,
+      },
+      [
+        {
+          $set: {
+            apiRateLimits: {
+              $mergeObjects: ['$apiRateLimits', apiRateLimits],
+            },
+          },
+        },
+      ]
+    );
+
+    return await this.getApiKeys(environmentId);
+  }
```

Comment:
This will be used to add custom rate limits per environment when enterprise customers need custom limits. It's also used for testing.